crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java

```diff
@@ -259,6 +277,50 @@ void shouldApplySchemaSwapsMultipleTimesInDeepClassHierarchy() {
     }
   }
 
+  @Test
+  void shouldApplyCyclicSchemaSwaps() {
```

Comment:
I think that here we are not testing that the substitution happened at the right depth:
given:
```java
@SchemaSwap(originalType = CyclicSchemaSwap.Level.class, fieldName = "level", depth = 0)
```
we should check something like:
```java
spec.get("level").get("level") == null
```

Reply:
I meant to leave this class using a depth of 1, that and the test have been updated.

Reply:
Thanks for updating the test, I do see some value in testing for N > 1, is it too much of a burden to add another check?

Reply:
ok 👍